base/util.jl

```diff
@@ -320,13 +320,26 @@ end
 function with_output_color(f::Function, color::Union{Int, Symbol}, io::IO, args...; bold::Bool = false)
     buf = IOBuffer()
     iscolor = get(io, :color, false)
-    iscolor && bold && print(buf, text_colors[:bold])
-    iscolor && print(buf, get(text_colors, color, color_normal))
     try f(IOContext(buf, io), args...)
     finally
-        iscolor && color != :nothing && print(buf, get(disable_text_style, color, text_colors[:default]))
-        iscolor && (bold || color == :bold) && print(buf, disable_text_style[:bold])
-        print(io, String(take!(buf)))
+        str = String(take!(buf))
+        if !iscolor
+            print(io, str)
+        else
+            lines = split(str, '\n')
+            first = true
+            for line in lines
+                first || print(buf, '\n')
+                first = false
+                isempty(line) && continue
+                bold && color != :bold && print(buf, text_colors[:bold])
```

Comment:
Seems better to just add a
```
bold && color == :bold && (color = :nothing)
```
line above.

Reply:
Alright, did this and shoved the dictionary lookups outside the loop.